dotnet: add feature bands

corngood opened the pull request:
## Description of changes

FYI @NixOS/dotnet 

This adds packages like `dotnetCorePackages.sdk_8_0_2xx` and `sdk_8_0_1xx` for all available feature bands, and points `sdk_8_0`.  I've also updated everything except the source-built 9.0 preview SDK, which needs substantial work.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

corngood committed:
Revert "dotnet/update.sh: use sdk patch version instead of runtime"

This reverts commit 845b559abe7dea12b2f3d4a98cf5326e0f1e195b.

corngood committed:
dotnet: add support for feature bands in update.sh

This will make e.g. both sdk_8_0_2xx and sdk_8_0_1xx available, with
sdk_8_0 poiting to the latest.

corngood committed:
dotnet: fix stripping on darwin

Fixes: 88e8d87fedcd17d7125d970fc2e6ff03c9884d6a

corngood committed:
dotnet-sdk_6: 6.0.27 -> 6.0.28

corngood committed:
dotnet-sdk_7: 7.0.16 -> 7.0.17

corngood committed:
dotnet-sdk_8: 8.0.2 -> 8.0.3

corngood committed:
dotnetCorePackages.dotnet_8: 8.0.2 -> 8.0.3

GGG-KILLER commented on pkgs/development/compilers/dotnet/update.sh:
```diff
@@ -390,17 +388,33 @@ Examples:
+    for sdk_version in "${sdk_versions[@]}"; do
+      sdk_base_version=${sdk_version%-*}
+      feature_band=${sdk_base_version:0:-2}xx
+      # sometimes one release has e.g. both 8.0.202 and 8.0.203
+      [[ ! ${feature_bands[$feature_band]+true} ]] || continue
```
Do we have a guarantee here that they're sorted in highest to lowest version? Or do we risk ending up with 8.0.202 in nixpkgs when 8.0.203 was available but placed later on the list of SDK versions?

corngood replied:
Yeah, it sorts them descending and then takes the first one for each feature level.

I'm not sure why both of those sdks are included in the 2.0.3 'release'.

GGG-KILLER commented:
Also, maybe possible future idea, for 8.0 we could have `runtime`, `aspnetcore`, and `sdk_8_0_1xx` point to the source built and the rest point to binary one

Or just wait for 9.0 and have every one of them be source built

GGG-KILLER commented:
Result of `nixpkgs-review pr 299822` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li>dotnetCorePackages.dotnet_9.aspnetcore</li>
    <li>dotnetCorePackages.dotnet_9.runtime</li>
    <li>dotnetCorePackages.dotnet_9.sdk</li>
    <li>dotnetCorePackages.dotnet_9.sdk.artifacts</li>
    <li>dotnetCorePackages.dotnet_9.sdk.packages</li>
    <li>dotnetCorePackages.dotnet_9.vmr</li>
    <li>roslyn-ls</li>
  </ul>
</details>
<details>
  <summary>122 packages built:</summary>
  <ul>
    <li>ArchiSteamFarm</li>
    <li>BeatSaberModManager</li>
    <li>alttpr-opentracker</li>
    <li>am2rlauncher</li>
    <li>audiobookshelf</li>
    <li>avalonia-ilspy</li>
    <li>azure-functions-core-tools</li>
    <li>bicep</li>
    <li>bililiverecorder</li>
    <li>boogie (dotnetPackages.Boogie)</li>
    <li>btcpayserver</li>
    <li>btcpayserver-altcoins</li>
    <li>cavalier</li>
    <li>celeste64</li>
    <li>certdump</li>
    <li>csharp-ls</li>
    <li>csharpier</li>
    <li>csharprepl</li>
    <li>dafny</li>
    <li>denaro</li>
    <li>depotdownloader</li>
    <li>discordchatexporter-cli</li>
    <li>dotnet-aspnetcore (dotnet-aspnetcore_6 ,dotnetCorePackages.aspnetcore_6_0)</li>
    <li>dotnet-aspnetcore_7 (dotnetCorePackages.aspnetcore_7_0)</li>
    <li>dotnet-aspnetcore_8 (dotnetCorePackages.aspnetcore_8_0)</li>
    <li>dotnet-runtime (dotnet-runtime_6 ,dotnetCorePackages.runtime_6_0)</li>
    <li>dotnet-runtime_7 (dotnetCorePackages.runtime_7_0)</li>
    <li>dotnet-runtime_8 (dotnetCorePackages.runtime_8_0)</li>
    <li>dotnet-sdk (dotnet-sdk_6 ,dotnetCorePackages.sdk_6_0 ,dotnetCorePackages.sdk_6_0_4xx)</li>
    <li>dotnet-sdk_7 (dotnetCorePackages.sdk_7_0 ,dotnetCorePackages.sdk_7_0_4xx)</li>
    <li>dotnet-sdk_8 (dotnetCorePackages.sdk_8_0 ,dotnetCorePackages.sdk_8_0_2xx)</li>
    <li>dotnetCorePackages.aspnetcore_9_0</li>
    <li>dotnetCorePackages.dotnet_8.aspnetcore</li>
    <li>dotnetCorePackages.dotnet_8.runtime</li>
    <li>dotnetCorePackages.dotnet_8.sdk</li>
    <li>dotnetCorePackages.dotnet_8.sdk.artifacts</li>
    <li>dotnetCorePackages.dotnet_8.sdk.packages</li>
    <li>dotnetCorePackages.dotnet_8.vmr</li>
    <li>dotnetCorePackages.runtime_9_0</li>
    <li>dotnetCorePackages.sdk_6_0_1xx</li>
    <li>dotnetCorePackages.sdk_7_0_1xx</li>
    <li>dotnetCorePackages.sdk_7_0_3xx</li>
    <li>dotnetCorePackages.sdk_8_0_1xx</li>
    <li>dotnetCorePackages.sdk_9_0 (dotnetCorePackages.sdk_9_0_1xx)</li>
    <li>eventstore</li>
    <li>fable</li>
    <li>famistudio</li>
    <li>fantomas</li>
    <li>formula</li>
    <li>fsautocomplete</li>
    <li>galaxy-buds-client</li>
    <li>git-credential-manager</li>
    <li>github-runner</li>
    <li>godot3-mono</li>
    <li>godot3-mono-debug-server</li>
    <li>godot3-mono-export-templates</li>
    <li>godot3-mono-headless</li>
    <li>godot3-mono-headless.dev</li>
    <li>godot3-mono-headless.man</li>
    <li>godot3-mono-server</li>
    <li>godot3-mono.dev</li>
    <li>godot3-mono.man</li>
    <li>ilspycmd</li>
    <li>inklecate</li>
    <li>jackett</li>
    <li>jellyfin</li>
    <li>jetbrains.rider</li>
    <li>kavita</li>
    <li>knossosnet</li>
    <li>libation</li>
    <li>lidarr</li>
    <li>lubelogger</li>
    <li>marksman</li>
    <li>mqttmultimeter</li>
    <li>msbuild</li>
    <li>naps2</li>
    <li>nbxplorer</li>
    <li>netcoredbg</li>
    <li>networkminer</li>
    <li>nuget-to-nix</li>
    <li>omnisharp-roslyn</li>
    <li>openra</li>
    <li>opentabletdriver</li>
    <li>openutau</li>
    <li>osu-lazer</li>
    <li>parabolic</li>
    <li>pbm</li>
    <li>pinta</li>
    <li>pre-commit</li>
    <li>pre-commit.dist</li>
    <li>prowlarr</li>
    <li>ps3-disc-dumper</li>
    <li>pupdate</li>
    <li>python311Packages.asteroid-filterbanks</li>
    <li>python311Packages.asteroid-filterbanks.dist</li>
    <li>python311Packages.clr-loader</li>
    <li>python311Packages.clr-loader.dist</li>
    <li>python311Packages.pyannote-audio</li>
    <li>python311Packages.pyannote-audio.dist</li>
    <li>python311Packages.pythonnet</li>
    <li>python311Packages.pythonnet.dist</li>
    <li>python312Packages.clr-loader</li>
    <li>python312Packages.clr-loader.dist</li>
    <li>python312Packages.pythonnet</li>
    <li>python312Packages.pythonnet.dist</li>
    <li>radarr</li>
    <li>readarr</li>
    <li>roslyn</li>
    <li>ryujinx</li>
    <li>scarab</li>
    <li>slskd</li>
    <li>sonarr</li>
    <li>space-station-14-launcher</li>
    <li>technitium-dns-server</li>
    <li>tests.dotnet.project-references</li>
    <li>tests.writers.simple.fsharp</li>
    <li>tests.writers.simple.fsharpNoNugetDeps</li>
    <li>tone</li>
    <li>torrentstream</li>
    <li>vintagestory</li>
    <li>wasabibackend</li>
    <li>xivlauncher</li>
  </ul>
</details>

GGG-KILLER commented:
Logs for the 2 build failures:
[dotnet-vmr-9.0.0-preview.1.log](https://github.com/NixOS/nixpkgs/files/14809108/dotnet-vmr-9.0.0-preview.1.log)
[roslyn-ls-4.10.0-2.24124.2.log](https://github.com/NixOS/nixpkgs/files/14809109/roslyn-ls-4.10.0-2.24124.2.log)

I'm checking if they're a result of this PR or not currently.

GGG-KILLER commented:
`roslyn-ls` built normally on `master`, could you take a look @corngood?

corngood commented:
> Also, maybe possible future idea, for 8.0 we could have runtime_8_0, aspnetcore_8_0, and sdk_8_0_1xx point to the source built and the rest point to binary one

Yeah, I'd like to do that.  The source-built `sdk.packages` still has a problem with cross-compilation, in that it only contains packages from the target 'rid', and the sdk itself doesn't build in cross at the moment.  At some point I'll do more work on cross compilation.  Perhaps we can split those packages up by rid and fall back to the binary ones when in cross.

> Result of nixpkgs-review pr 299822 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

Thanks for testing.  I'll take a look at `roslyn-ls`.  The `dotnet_9` failures are probably just from building preview 1 on preview 2.  I think I'll drop the binary upgrade of 9.0 from this PR while I work on getting 9.0 preview 2 to compile from source.

corngood committed:
roslyn-ls: build with EnableWindowsTargeting=false

This removes the dependency on Microsoft.WindowsDesktop.App.Ref, which
is tied to the runtime version. We can't trivially add it to the
standard list of sdk packages, because its version doesn't exactly match
either the runtime or sdk versions. For example: 9.0.0-preview.1.24081.3
is needed for runtime 9.0.0-preview.1.24081.5.

Since we don't currently support windows targets, we'll just disable
this for now.

This will allow the sdks to be updated without changing the deps file.

corngood commented:
@konradmalik could you take a look at the `roslyn-ls` change here?  It should allow it to be built on new SDKs without deps changing.

corngood commented:
Result of `nixpkgs-review pr 299822` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>dotnet-outdated</li>
  </ul>
</details>
<details>
  <summary>117 packages built:</summary>
  <ul>
    <li>ArchiSteamFarm</li>
    <li>BeatSaberModManager</li>
    <li>alttpr-opentracker</li>
    <li>am2rlauncher</li>
    <li>audiobookshelf</li>
    <li>avalonia-ilspy</li>
    <li>azure-functions-core-tools</li>
    <li>bicep</li>
    <li>bililiverecorder</li>
    <li>boogie</li>
    <li>btcpayserver</li>
    <li>btcpayserver-altcoins</li>
    <li>cavalier</li>
    <li>celeste64</li>
    <li>certdump</li>
    <li>csharp-ls</li>
    <li>csharpier</li>
    <li>csharprepl</li>
    <li>dafny</li>
    <li>denaro</li>
    <li>depotdownloader</li>
    <li>discordchatexporter-cli</li>
    <li>dotnet-aspnetcore</li>
    <li>dotnet-aspnetcore_7</li>
    <li>dotnet-aspnetcore_8</li>
    <li>dotnet-runtime</li>
    <li>dotnet-runtime_7</li>
    <li>dotnet-runtime_8</li>
    <li>dotnet-sdk</li>
    <li>dotnet-sdk_7</li>
    <li>dotnet-sdk_8</li>
    <li>dotnetCorePackages.dotnet_8.aspnetcore</li>
    <li>dotnetCorePackages.dotnet_8.runtime</li>
    <li>dotnetCorePackages.dotnet_8.sdk</li>
    <li>dotnetCorePackages.dotnet_8.sdk.artifacts</li>
    <li>dotnetCorePackages.dotnet_8.sdk.packages</li>
    <li>dotnetCorePackages.dotnet_8.vmr</li>
    <li>dotnetCorePackages.sdk_6_0_1xx</li>
    <li>dotnetCorePackages.sdk_7_0_1xx</li>
    <li>dotnetCorePackages.sdk_7_0_3xx</li>
    <li>dotnetCorePackages.sdk_8_0_1xx</li>
    <li>eventstore</li>
    <li>fable</li>
    <li>famistudio</li>
    <li>fantomas</li>
    <li>formula</li>
    <li>fsautocomplete</li>
    <li>galaxy-buds-client</li>
    <li>git-credential-manager</li>
    <li>github-runner</li>
    <li>godot3-mono</li>
    <li>godot3-mono-debug-server</li>
    <li>godot3-mono-export-templates</li>
    <li>godot3-mono-headless</li>
    <li>godot3-mono-headless.dev</li>
    <li>godot3-mono-headless.man</li>
    <li>godot3-mono-server</li>
    <li>godot3-mono.dev</li>
    <li>godot3-mono.man</li>
    <li>ilspycmd</li>
    <li>inklecate</li>
    <li>jackett</li>
    <li>jellyfin</li>
    <li>jetbrains.rider</li>
    <li>kavita</li>
    <li>knossosnet</li>
    <li>libation</li>
    <li>lidarr</li>
    <li>lubelogger</li>
    <li>marksman</li>
    <li>mqttmultimeter</li>
    <li>msbuild</li>
    <li>naps2</li>
    <li>nbxplorer</li>
    <li>netcoredbg</li>
    <li>networkminer</li>
    <li>nuget-to-nix</li>
    <li>omnisharp-roslyn</li>
    <li>openra</li>
    <li>opentabletdriver</li>
    <li>openutau</li>
    <li>osu-lazer</li>
    <li>parabolic</li>
    <li>pbm</li>
    <li>pinta</li>
    <li>pre-commit</li>
    <li>pre-commit.dist</li>
    <li>prowlarr</li>
    <li>ps3-disc-dumper</li>
    <li>pupdate</li>
    <li>python311Packages.asteroid-filterbanks</li>
    <li>python311Packages.asteroid-filterbanks.dist</li>
    <li>python311Packages.clr-loader</li>
    <li>python311Packages.clr-loader.dist</li>
    <li>python311Packages.pyannote-audio</li>
    <li>python311Packages.pyannote-audio.dist</li>
    <li>python311Packages.pythonnet</li>
    <li>python311Packages.pythonnet.dist</li>
    <li>python312Packages.clr-loader</li>
    <li>python312Packages.clr-loader.dist</li>
    <li>python312Packages.pythonnet</li>
    <li>python312Packages.pythonnet.dist</li>
    <li>radarr</li>
    <li>readarr</li>
    <li>roslyn</li>
    <li>roslyn-ls</li>
    <li>ryujinx</li>
    <li>scarab</li>
    <li>slskd</li>
    <li>sonarr</li>
    <li>space-station-14-launcher</li>
    <li>technitium-dns-server</li>
    <li>tone</li>
    <li>torrentstream</li>
    <li>vintagestory</li>
    <li>wasabibackend</li>
    <li>xivlauncher</li>
  </ul>
</details>

corngood committed:
dotnet-outdated: build with binary sdks 6/7/8

This removes all the sdk-specific dependencies from deps.nix.

corngood commented:
@emilioziniades could you take a look at the `dotnet-outdated` change (6830aac1a9430e903c102d8c546b84a11ac49b21).  I switched it to use a composite of the 6/7/8 binary SDKs, which removes all the sdk/runtime dependent stuff from deps.nix.

corngood commented:
Result of `nixpkgs-review pr 299822` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>118 packages built:</summary>
  <ul>
    <li>ArchiSteamFarm</li>
    <li>BeatSaberModManager</li>
    <li>alttpr-opentracker</li>
    <li>am2rlauncher</li>
    <li>audiobookshelf</li>
    <li>avalonia-ilspy</li>
    <li>azure-functions-core-tools</li>
    <li>bicep</li>
    <li>bililiverecorder</li>
    <li>boogie</li>
    <li>btcpayserver</li>
    <li>btcpayserver-altcoins</li>
    <li>cavalier</li>
    <li>celeste64</li>
    <li>certdump</li>
    <li>csharp-ls</li>
    <li>csharpier</li>
    <li>csharprepl</li>
    <li>dafny</li>
    <li>denaro</li>
    <li>depotdownloader</li>
    <li>discordchatexporter-cli</li>
    <li>dotnet-aspnetcore</li>
    <li>dotnet-aspnetcore_7</li>
    <li>dotnet-aspnetcore_8</li>
    <li>dotnet-outdated</li>
    <li>dotnet-runtime</li>
    <li>dotnet-runtime_7</li>
    <li>dotnet-runtime_8</li>
    <li>dotnet-sdk</li>
    <li>dotnet-sdk_7</li>
    <li>dotnet-sdk_8</li>
    <li>dotnetCorePackages.dotnet_8.aspnetcore</li>
    <li>dotnetCorePackages.dotnet_8.runtime</li>
    <li>dotnetCorePackages.dotnet_8.sdk</li>
    <li>dotnetCorePackages.dotnet_8.sdk.artifacts</li>
    <li>dotnetCorePackages.dotnet_8.sdk.packages</li>
    <li>dotnetCorePackages.dotnet_8.vmr</li>
    <li>dotnetCorePackages.sdk_6_0_1xx</li>
    <li>dotnetCorePackages.sdk_7_0_1xx</li>
    <li>dotnetCorePackages.sdk_7_0_3xx</li>
    <li>dotnetCorePackages.sdk_8_0_1xx</li>
    <li>eventstore</li>
    <li>fable</li>
    <li>famistudio</li>
    <li>fantomas</li>
    <li>formula</li>
    <li>fsautocomplete</li>
    <li>galaxy-buds-client</li>
    <li>git-credential-manager</li>
    <li>github-runner</li>
    <li>godot3-mono</li>
    <li>godot3-mono-debug-server</li>
    <li>godot3-mono-export-templates</li>
    <li>godot3-mono-headless</li>
    <li>godot3-mono-headless.dev</li>
    <li>godot3-mono-headless.man</li>
    <li>godot3-mono-server</li>
    <li>godot3-mono.dev</li>
    <li>godot3-mono.man</li>
    <li>ilspycmd</li>
    <li>inklecate</li>
    <li>jackett</li>
    <li>jellyfin</li>
    <li>jetbrains.rider</li>
    <li>kavita</li>
    <li>knossosnet</li>
    <li>libation</li>
    <li>lidarr</li>
    <li>lubelogger</li>
    <li>marksman</li>
    <li>mqttmultimeter</li>
    <li>msbuild</li>
    <li>naps2</li>
    <li>nbxplorer</li>
    <li>netcoredbg</li>
    <li>networkminer</li>
    <li>nuget-to-nix</li>
    <li>omnisharp-roslyn</li>
    <li>openra</li>
    <li>opentabletdriver</li>
    <li>openutau</li>
    <li>osu-lazer</li>
    <li>parabolic</li>
    <li>pbm</li>
    <li>pinta</li>
    <li>pre-commit</li>
    <li>pre-commit.dist</li>
    <li>prowlarr</li>
    <li>ps3-disc-dumper</li>
    <li>pupdate</li>
    <li>python311Packages.asteroid-filterbanks</li>
    <li>python311Packages.asteroid-filterbanks.dist</li>
    <li>python311Packages.clr-loader</li>
    <li>python311Packages.clr-loader.dist</li>
    <li>python311Packages.pyannote-audio</li>
    <li>python311Packages.pyannote-audio.dist</li>
    <li>python311Packages.pythonnet</li>
    <li>python311Packages.pythonnet.dist</li>
    <li>python312Packages.clr-loader</li>
    <li>python312Packages.clr-loader.dist</li>
    <li>python312Packages.pythonnet</li>
    <li>python312Packages.pythonnet.dist</li>
    <li>radarr</li>
    <li>readarr</li>
    <li>roslyn</li>
    <li>roslyn-ls</li>
    <li>ryujinx</li>
    <li>scarab</li>
    <li>slskd</li>
    <li>sonarr</li>
    <li>space-station-14-launcher</li>
    <li>technitium-dns-server</li>
    <li>tone</li>
    <li>torrentstream</li>
    <li>vintagestory</li>
    <li>wasabibackend</li>
    <li>xivlauncher</li>
  </ul>
</details>

emilioziniades commented:
> @emilioziniades could you take a look at the `dotnet-outdated` change (6830aac1a9430e903c102d8c546b84a11ac49b21).  I switched it to use a composite of the 6/7/8 binary SDKs, which removes all the sdk/runtime dependent stuff from deps.nix.

LGTM

> Using the source-built sdk causes fetch-deps to add some dependencies, which I'll need to investigate.

Thanks, I didn't realize there were binary sdks and source-built ones. Is that documented somewhere?

GGG-KILLER commented:
> > Using the source-built sdk causes fetch-deps to add some dependencies, which I'll need to investigate.
> 
> Thanks, I didn't realize there were binary sdks and source-built ones. Is that documented somewhere?

It is a recent effort by @corngood, long term the objective is to use only source build so we stop having issues due to nix quirks.

The difference is that `{sdk,runtime,aspect}_X_Y` *currently* are the binary releases from MS but patched to work in Nix and the `dotnet_X` are the source built releases.

Currently we have source builds for 8.0.1xx and 9.0 thanks to the wonderful efforts by @corngood

konradmalik commented:
> @konradmalik could you take a look at the `roslyn-ls` change here? It should allow it to be built on new SDKs without deps changing.

Change looks ok. Also I've built and tested the package - works ok. Thanks!
